x/storage/keeper/msg_server_buy_storage.go

```diff
@@ -76,12 +75,25 @@ func (k msgServer) BuyStorage(goCtx context.Context, msg *types.MsgBuyStorage) (
 		return nil, err
 	}
 
-	spi := types.StoragePaymentInfo{
-		Start:          ctx.BlockTime(),
-		End:            ctx.BlockTime().Add(dm),
-		SpaceAvailable: bytes,
-		SpaceUsed:      0,
-		Address:        msg.ForAddress,
+	var spi types.StoragePaymentInfo
+
+	payInfo, found := k.GetStoragePaymentInfo(ctx, msg.ForAddress)
+	if found {
+		spi = types.StoragePaymentInfo{
+			Start:          ctx.BlockTime(),
+			End:            ctx.BlockTime().Add(dm),
+			SpaceAvailable: bytes,
+			SpaceUsed:      payInfo.SpaceUsed,
```

Comment:
@nuggetnchill can we make this so that if the spaceUsed is going to be greater than the bytes they requested to not allow it?